### PR TITLE
remove trusted service url check in appcredentials

### PR DESCRIPTION
**Pull request by johnataylor**
This PR removed the check on Trusted Service Url in the custom AppCredentials class.

Some of the internal implementation has been removed and the public methods obsoleted.

Tests have been updated and removed. (Also note the odd behavior in the TeamsInfo test. It seems there is a subtle distinction between null and empty string in no-auth scenarios.)

This check is redundant as all the logic does is add to a collection and then a couple of stack frames later check for the presence of the value in that same collection.

This was never a security feature exactly. It was a feature added in V3 to stop accidental deployment of code to production that didn't contain the ASP MVC auth attribute. In v4 the inbound / outbound nature of the protocol is fully encapsulated with the adapter and it does not rely on the user adding attributes to a ASP MVC controller.  

This code had more confusion in the world of Skills because the ContinueConversation implementation also needs to add to this collection just to get the lower level function to behave.

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsInfoTests.cs
Comment (stevengum) [inline tests/Microsoft.Bot.Builder.Tests/Teams/TeamsInfoTests.cs:305]:
```suggestion
                var creds = MicrosoftAppCredentials.Empty;
```

File: libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:22]: <3

File: tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterTests.cs
Comment (carlosscastro) [inline tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterTests.cs:309]: Why not delete?

File: libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs
Comment (stevengum) [inline libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:238]: This method is now unused, it can be removed

**Review by carlosscastro — APPROVED** [review]

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsInfoTests.cs
Comment (johnataylor) [reply tests/Microsoft.Bot.Builder.Tests/Teams/TeamsInfoTests.cs:305]: surprisingly this doesn't work everywhere - the create conversation call needs new MicrosoftAppCredentials(string.Empty, string.Empty) otherwise you get a null exception. We can fix that one in Cloud Adapter

File: libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs
Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:238]: well spotted - next commit!

File: tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterTests.cs
Comment (johnataylor) [reply tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterTests.cs:309]: yes absolutely! thanks!

**Comment by johnataylor** [comment]
Thanks for the reviews. I pushed another commit, but I'll do a little more local testing before merging.

And then as a follow on PR I'm planning on an update to enable continue conversation (aka proactive) in CloudAdapter.

**Comment by johnataylor** [comment]
And we can do parity in JS, Python and Java in the next release. This unblock Cloud Adapter which we only have in C# so far.

**Review by gabog — APPROVED** [review]
Looks good, we just need to add a description to the PR explaining what the changes are about and also create an issue to mark all these methods as onsolete and update docs in R12

**Review by stevengum — APPROVED** [review]
:shipit:

**Pull request merged**
